deepchem/models/torch_models/cgcnn.py

```diff
@@ -0,0 +1,192 @@
+  ----------
+  batch: Tuple
+    The tuple are inputs, labels, weights. The inputs has 
+
+  device: torch.device
+    the device on which to run computations.  If None, a device is
+    chosen automatically.
+
+  Returns
+  -------
+
+
+  Notes
+  -----
+  This class requires DGL and PyTorch to be installed.
+  """
+  try:
+    import dgl
+  except:
+    raise ValueError("This class requires DGL to be installed.")
+
+  inputs, labels, weights = batch
+  inputs = dgl.batch([graph.to_dgl_graph() for graph in inputs[0]]).to(device)
+  if labels is not None:
+    labels = [
+        x.astype(np.float32) if x.dtype == np.float64 else x for x in labels
+    ]
+    labels = [torch.as_tensor(x, device=device) for x in labels]
+  if weights is not None:
+    weights = [
+        x.astype(np.float32) if x.dtype == np.float64 else x for x in weights
+    ]
+    weights = [torch.as_tensor(x, device=device) for x in weights]
+
+  return inputs, labels, weights
+
+
+class CGCNNLayer(nn.Module):
+  """The convolutional layer of CGCNN.
+
+  This class was implemented using DGLGraph methods.
+
+  See: https://docs.dgl.ai/en/0.4.x/tutorials/models/1_gnn/9_gat.html
+  """
+
+  def __init__(self,
+               hidden_node_dim: int,
+               edge_dim: int,
+               batch_norm: bool = True):
+    """
+    Parameters
+    ----------
+    hidden_node_dim: int
+      The length of the hidden node feature vectors.
+    edge_dim: int
+      The length of the edge feature vectors.
+    batch_norm: bool, default True
+      Whether to apply batch normalization or not.
+    """
+    super(CGCNNLayer, self).__init__()
+    z_dim = 2 * hidden_node_dim + edge_dim
+    self.linear_with_sigmoid = nn.Linear(z_dim, hidden_node_dim)
+    self.linear_with_softplus = nn.Linear(z_dim, hidden_node_dim)
+    self.batch_norm = nn.BatchNorm1d(hidden_node_dim) if batch_norm else None
+
+  def message_func(self, edges):
+    z = torch.cat(
+        [edges.src['x'], edges.dst['x'], edges.data['edge_attr']], dim=1)
+    gated_z = F.sigmoid(self.linear_with_sigmoid(z))
+    message_z = F.softplus(self.linear_with_softplus(z))
+    return {'gated_z': gated_z, 'message_z': message_z}
+
+  def reduce_func(self, nodes):
+    new_h = nodes.data['x'] + torch.sum(
+        nodes.mailbox['gated_z'] * nodes.mailbox['message_z'], dim=1)
+    return {'x': new_h}
+
+  def forward(self, dgl_graph):
+    """Update node representaions.
+
+    Parameters
+    ----------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of graphs. The graph expects that the node features
+      are stored in `ndata['x']`, and the edge features are stored in `edata['edge_attr']`.
+
+    Returns
+    -------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of updated graphs.
+    """
+    dgl_graph.update_all(self.message_func, self.reduce_func)
+    if self.batch_norm is not None:
+      dgl_graph.ndata['x'] = self.batch_norm(dgl_graph.ndata['x'])
+    return dgl_graph
+
+
+class CGCNN(nn.Module):
```

Comment:
I think it would be better if we provide a class that extends TorchModel and internally creates the `torch.nn.Module`.  That's how all our Keras based models work.

Reply:
Is there any reason...?  I think this style is more familiar to many PyTorch users.

Reply:
But less familiar to DeepChem users.  This is how all our models work.  Classes like GraphConvModel and WeaveModel extend KerasModel.  We don't require users to manually wrap them in KerasModel.

Reply:
OK, I made CGCNNModel class